File: 백준/Bronze/2747. 피보나치 수/피보나치 수.py
import sys

input = sys.stdin.readline


def fib(num):
    if dp[num] == -1:  # 한번도 연산된 적이 없다면
        dp[num] = fib(num - 1) + fib(num - 2)
    return dp[num]


num = int(input())
dp = [-1] * 46
dp[0] = 0
dp[1] = 1
print(fib(num))

File: 백준/Bronze/2747. 피보나치 수/피보나치 수.kt
package org.example

fun main() {
    val br = System.`in`.bufferedReader()
    val num = br.readLine().toInt()
    val dp = IntArray(46)
    dp[0] = 0
    dp[1] = 1
    for (i in 2 until num + 1) {
        dp[i] = dp[i - 1] + dp[i - 2]
    }
    println(dp[num])
    br.close()
}
